core/tests/protractor/learnerFlow.js

```diff
@@ -0,0 +1,411 @@
+    explorationEditorPage.navigateToSettingsTab();
+    explorationEditorSettingsTab.setTitle('About Oppia');
+    explorationEditorSettingsTab.setCategory('Algorithm');
+    explorationEditorSettingsTab.setObjective('Learn more about Oppia');
+    explorationEditorSettingsTab.setLanguage('English');
+    explorationEditorPage.saveChanges();
+    workflow.publishExploration();
+  };
+
+  var checkIncompleteExplorationSection = function(explorationTitle) {
+    learnerDashboardPage.navigateToInCompleteSection();
+    learnerDashboardPage.navigateToIncompleteExplorationsSection();
+    learnerDashboardPage.expectTitleOfExplorationSummaryTileToMatch(
+      explorationTitle);
+  };
+
+  var checkCompleteExplorationSection = function(explorationTitle) {
+    learnerDashboardPage.navigateToCompletedSection();
+    learnerDashboardPage.navigateToCompletedExplorationsSection();
+    learnerDashboardPage.expectTitleOfExplorationSummaryTileToMatch(
+      explorationTitle);
+  };
+
+  var checkIncompleteCollectionSection = function(collectionTitle) {
+    learnerDashboardPage.navigateToInCompleteSection();
+    learnerDashboardPage.navigateToIncompleteCollectionsSection();
+    learnerDashboardPage.expectTitleOfCollectionSummaryTileToMatch(
+      collectionTitle);
+  };
+
+  var checkCompleteCollectionSection = function(collectionTitle) {
+    learnerDashboardPage.navigateToCompletedSection();
+    learnerDashboardPage.navigateToCompletedCollectionsSection();
+    learnerDashboardPage.expectTitleOfCollectionSummaryTileToMatch(
+      collectionTitle);
+  };
+
+  beforeAll(function() {
+    adminPage = new AdminPage.AdminPage();
+    libraryPage = new LibraryPage.LibraryPage();
+    learnerDashboardPage = new LearnerDashboardPage.LearnerDashboardPage();
+
+    if (browser.isMobile) {
+      var ADM_VISITOR = 'admVisitor';
+      users.createAndLoginAdminUserMobile(
+        'admVisitor@learner.com', ADM_VISITOR);
+      // Load /explore/24
+      adminPage.reloadExploration('learner_flow_test.yaml');
+      // Load /explore/22
+      adminPage.reloadExploration('protractor_mobile_test_exploration.yaml');
+      // Load /collection/1
+      adminPage.reloadCollection(1);
+      users.logout();
+      var LEARNER_USERNAME = 'learnerMobile';
+      users.createUser('learnerMobile@learnerFlow.com', LEARNER_USERNAME);
+    } else {
+      collectionEditorPage = new CollectionEditorPage.CollectionEditorPage();
+      creatorDashboardPage = new CreatorDashboardPage.CreatorDashboardPage();
+      explorationEditorPage = new ExplorationEditorPage.ExplorationEditorPage();
+      explorationEditorMainTab = explorationEditorPage.getMainTab();
+      explorationEditorSettingsTab = explorationEditorPage.getSettingsTab();
+      explorationPlayerPage = new ExplorationPlayerPage.ExplorationPlayerPage();
+    }
+  });
+
+  it('displays incomplete and completed explorations', function() {
+    if (browser.isMobile) {
```

Comment:
The branching is too much, here. Can you unify the tests more? You can make a dummy exp on desktop that's similar to the exp you're reloading on mobile, at least for the parts that you care about testing. Only the exp ID will be different, but you would then be testing the same things (incomplete, complete, delete, etc.) for functionality that exists in both versions. For functionality that only exists in the desktop version, that test should go to protractor_desktop (and similarly for mobile).

It is fine to rewrite the tests as needed to unify them (you don't have to be restricted to the exp IDs that were being used before), as long as you're still testing the same things as you were before. If the desktop/mobile tests differ in what they cover, pick the superset (aim for maximal coverage).

Reply:
When Apurv first approached me about reinstating the reloadExploration() and reloadCollection() in AdminPage, I did suggested him to create exp on Desktop mode and then switched to Mobile mode as I was on edge about putting back deleted codes.

1) I don't know how feasible my suggestion that since I don't have the setup for BrowserStack to actually do some kind of P.O.C. 

2) I did not have the full knowledge as to what he was working with, his exact needs. He even stated he needed to complete his milestone!! 

The result: So I just showed him the deleted code and went on with my day. 

Now that everyone is on the same page and progress are made. 
(Big) __IF__ it is possible to switch from Desktop to Mobile with Browserstack, we would not need the whole (DEV_MODE) and reload stuff in AdminPage.

Otherwise, this branching for loading exp and collection is a by-product of the reinstating deleted code in AdminPage.

Hope my info dump makes sense.

Reply:
I think I can incorporate the method as suggested by Sean [here](https://github.com/oppia/oppia/pull/5340#discussion_r205986478). Thanks!

Reply:
Done, branched at only necessary stages and unified all the tests. Also, playerFlow has been integrated with learnerFlow now.